src/main/java/net/minestom/server/instance/block/rule/vanilla/StairsPlacementRule.java

```diff
@@ -31,14 +31,13 @@ public Block blockPlace(@NotNull Instance instance,
         Facing facing = this.getFacing(player);
         Shape shape = this.getShape(instance, blockPosition, facing);
         BlockFace half = BlockFace.BOTTOM; // waiting for new block faces to be implemented
-        String waterlogged = "false"; // waiting for water to be implemented
+        Boolean waterlogged = instance.getBlock(blockPosition.relative(blockFace)).equals(Block.WATER);
 
         return block.withProperties(Map.of(
-                "facing", facing.toString(),
-                "half", half.toString(),
-                "shape", shape.toString(),
-                "waterlogged", waterlogged));
-
+                "facing", facing.name().toLowerCase(),
+                "half", half.name().toLowerCase(),
+                "shape", shape.name().toLowerCase(),
+                "waterlogged", String.valueOf(waterlogged)));
```

Comment:
Since you're using a Boolean object already, I'd recommend:
```suggestion
                "waterlogged", waterlogged.toString());
```